### PR TITLE
[updated] improve softmax implementation

### PR DESCRIPTION
**Pull request by gabrieldemarmiesse**
### Summary

This PR is an update of #10027 which I kinda broke. Sorry about it.

### PR Overview

I moved the correctness test in the backend tests. I think it belongs there rather than in the activations tests.

- [ ] This PR requires new unit tests [y/n] (make sure tests are included)
- [ ] This PR requires to update the documentation [y/n] (make sure the docs are up-to-date)
- [x] This PR is backwards compatible [y/n]
- [ ] This PR changes the current API [y/n] (all API changes need to be approved by fchollet)

### CODE REVIEW

File: tests/keras/activations_test.py [hunk 1]
```diff
@@ -79,6 +79,16 @@ def test_softmax_invalid():
...
 
 
+def test_softmax_3d():
+    """Test using a reference implementation of softmax.
+    """
+    x = K.placeholder(ndim=3)
+    f = K.function([x], [activations.softmax(x, axis=1)])
+    test_values = get_standard_values()[:, :, np.newaxis].copy()
+
+    result = f([test_values])[0]
```
Comment (fchollet) [inline tests/keras/activations_test.py]: Please add a correctness check on `result` values. You can 1) use random values for `test_values`, 2) use `np.exp`/etc to compute the expected results.

Comment (gabrieldemarmiesse) [reply tests/keras/activations_test.py]: Wouldn't that be redundant since I've already added the correctness tests of the `softmax` operation in the `backend_test.py`  in this PR?

Comment (fchollet) [reply tests/keras/activations_test.py]: It's cheap, and most other tests in this file have at least basic correctness checks. You could just check sums to 1.

Comment (gabrieldemarmiesse) [reply tests/keras/activations_test.py]: Ok sure thing.
